### PR TITLE
Fix compiletest: don't abbreviate the output

### PR DESCRIPTION
**Pull request by fzaiser**
### Description of changes: 

This just cost me three hours of my life. Apparently, for reasons that are beyond me, compiletest "abbreviates" the output, i.e. it removes the middle part. This causes problems with big tests (such as `tests/expected/trait-receiver/receiver_types.rs`), which fail nondeterministically, depending on whether the expected part of the output falls into the skipped middle part.

In this PR, I change it such that compiletest always looks at the whole output.

### Testing:

* How is this change tested? I came across this problem in #1310, so it will be tested by that one, once it's merged.

* Is this a refactor change? No

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: tools/compiletest/src/read2.rs [hunk 1]
```diff
@@ -6,77 +6,29 @@
 // FIXME: This is a complete copy of `cargo/src/cargo/util/read2.rs`
```
Comment (sanjit-bhat) [inline tools/compiletest/src/read2.rs:6]: I looked at the latest Cargo util directory, https://github.com/rust-lang/cargo/tree/master/src/cargo/util, and there doesn't seem to be any file named `read2.rs` anymore. Is this copied code outdated? Should we consider updating it?

**Review by sanjit-bhat — CHANGES_REQUESTED** [review]
All changes LGTM. I've tested them locally with my PR, #1356, and they cause the regression to pass. I also manually changed one of the expected files to add gibberish, and as expected, that test fails.

File: tools/compiletest/src/read2.rs [hunk 2]
```diff
@@ -6,77 +6,29 @@
...
 // Consider unify the read2() in libstd, cargo and this to prevent further code duplication.
 
-pub use self::imp::read2;
+// Note: read2_abbreviated was replaced by read2 because the original function skipped some of the output,
+// which caused issues with our tests (in particular tests with large outputs).
```
Comment (adpaco-aws) [inline tools/compiletest/src/read2.rs:10]: Appreciate adding this comment, but if you remove `read2_abbreviated` it's surely going to confuse whoever works on this file in the future. Three ways to approach this:
 * Remove `read2_abbreviated` and don't add the comment.
 * Keep `read2_abbreviated` and add the comment for comparison.
 * Remove `read2_abbreviated` and add a comment in new line 90 in `tools/compiletest/src/runtest.rs` that refers to it (explaining it shortened the output) and explain that this is not desirable behavior, so you just the full output instead.

**Review by adpaco-aws — CHANGES_REQUESTED** [review]
Looks good but had some notes about the comments.

File: tools/compiletest/src/read2.rs (hunk 2 above)
Comment (fzaiser) [reply tools/compiletest/src/read2.rs:10]: I decided to remove `read2_abbreviated` and the comment.

File: tools/compiletest/src/read2.rs (hunk 1 above)
Comment (fzaiser) [reply tools/compiletest/src/read2.rs:6]: I removed the comment.

**Review by adpaco-aws — APPROVED** [review]

**Review by sanjit-bhat — APPROVED** [review]

**Pull request merged**